### ISSUE TITLE
[DocDB] Add ability to specify an RPC connect timeout

### ISSUE THREAD

**Issue by tylarb** [yugabyte/yugabyte-db#11306]
### Description

In a scenario where a network just drops all incoming packets to an offline node (certain on-prem providers), hypothetically:

1) a node is removed from a universe (via platform)
2) A tserver sends a Read RPC to the offline node from its meta cache
3) The tserver attempts to connect to the offline node and the `connect()` times out
4) The RPC deadline is reached
5) The RPC fails with `SQL Error: Execution Error. Read RPC (request call id 774107089) to xxx.xxx.xxx.xxx:9100 timed out after 5.000s`. This is a query failure, instead of a silent retry after refreshing the meta cache.

Notes:
- the query can be retried successfully (the cache is refreshed on RPC timeout)
- this will happen at most once per tablet per server - the meta cache is stored for each tablet, on each server. 
- a restart of the tserver will also refresh the meta cache
- However, this has a bit of emotional concern, because the meta cache is not refreshed automatically. Thus, it's possible that we get a query failed with a message that points to a node which hasn't been part of the cluster for hours or days (and, could in theory be longer). So while the impact is low, the appearance is concerning.

We have a `connection` timeout - for `send()` and `recv()` - clarified in this comment: https://github.com/yugabyte/yugabyte-db/issues/950#issuecomment-470616133  controlled with the flag `rpc_connection_timeout_ms`

But we don't have a `connect()` timeout - a timeout on forming a _new_ connection.

In testing, if `tserver_yb_client_default_timeout_ms` is set to lower than the OS default `connect()` timeout which is about 2 minutes, this scenario can happen.

One workaround is to tune the OS setting - there's not a global sysctl for `connect()` timeout, but one can set `/proc/sys/net/ipv4/tcp_syn_retries` - by default, this is `6` on RHEL derivatives, which would give a timeout of about 130 seconds (see explanation here: https://blog.cloudflare.com/when-tcp-sockets-refuse-to-die/)

So, this condition won't be hit if `/proc/sys/net/ipv4/tcp_syn_retries` is set to something low enough that `connect()` times out faster than RPC timeout. In testing, an RPC timeout of 15 seconds requires `tcp_syn_retries` set to `2` - 10 second total retry. For RPC timeout of 5 seconds, `1` works - 3 second total retry.

Two proposals to address this from a DB flag:

1) Make `rpc_connection_timeout_ms` also affects `connect()` timeout
2) Add a new flag, perhaps `rpc_connect_timeout_ms` that controls just new connections, not existing connections like `rpc_connection_timeout_ms` . This flag could be `-1` (use OS setting) by default, but if it is set, it adds `TCP_USER_TIMEOUT` socket setting to timeout a new `connect()`

**Comment by amitanandaiyer** [comment]
after https://github.com/yugabyte/yugabyte-db/issues/11799 we already clear out a bad node from the quorum upon any connect-failure (from any tablet) so we should already be fixing the issue that the customer has seen.

In theory, though, we could still run into this scenario if we have never tried *any* rpc to the bad node after it has been removed. But the very first rpc (to any tablet) addressed to the bad node will discover that it is unreachable and clear it out from meta-cache for all nodes.

So tldr; the probability of running into this has gone down from requiring that at least one tablet was not touched to requiring that none of the tablets hosted on the bad node was accessed after the removal. which should be really small. However, in theory, we could orchestrate a similar scenario.
